wire-compiler/src/main/java/com/squareup/wire/CommandLineOptions.java

```diff
@@ -0,0 +1,137 @@
+package com.squareup.wire;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Set;
+
+
+final class CommandLineOptions {
+  public static final String PROTO_PATH_FLAG = "--proto_path=";
+  public static final String JAVA_OUT_FLAG = "--java_out=";
+  public static final String FILES_FLAG = "--files=";
+  public static final String REGISTRY_CLASS_FLAG = "--registry_class=";
+  public static final String ROOTS_FLAG = "--roots=";
+  public static final String NO_OPTIONS_FLAG = "--no_options";
+  public static final String ENUM_OPTIONS_FLAG = "--enum_options=";
+  public static final String SERVICE_WRITER_FLAG = "--service_writer=";
+  public static final String SERVICE_WRITER_OPT_FLAG = "--service_writer_opt=";
+
+  final String protoPath;
+  final List<String> sourceFileNames;
+  final List<String> roots;
+  final String javaOut;
+  final String registryClass;
+  final boolean emitOptions;
+  final Set<String> enumOptions;
+  final String serviceWriter;
+  final List<String> serviceWriterOptions;
+
+  CommandLineOptions(String protoPath, List<String> sourceFileNames, List<String> roots,
```

Comment:
This is a potential place to use [@AutoValue](https://github.com/google/auto) again.   I don't think it adds a lot of value in this case either, but if you are gung ho to use this library, let me know and I will add it here and in the OutputArtifact case in PR 180.